subsys/nrf_security/src/drivers/cracen/cracenpsa/src/spake2p.c

```diff
@@ -224,10 +222,33 @@ static psa_status_t cracen_get_confirmation_keys(cracen_spake2p_operation_t *ope
 		return status;
 	}
 
-	/* Get K_e (K_shared) */
-	__ASSERT_NO_MSG(hash_len / 2 == half_hash);
-	/* Second half the hash is used. */
-	memcpy(operation->shared, V + half_hash, half_hash);
+	if (operation->alg == PSA_ALG_SPAKE2P_MATTER) {
+		operation->shared_len = hash_len / 2;
+		memcpy(operation->shared, V + operation->shared_len, operation->shared_len);
+	} else {
+		operation->shared_len = hash_len;
+		status = psa_driver_wrapper_key_derivation_setup(&kdf_op,
+								 PSA_ALG_HKDF(PSA_ALG_SHA_256));
+		if (status) {
+			goto exit;
+		}
+		status = psa_driver_wrapper_key_derivation_input_bytes(
+			&kdf_op, PSA_KEY_DERIVATION_INPUT_INFO, (uint8_t *)"SharedKey", 9);
+		if (status) {
+			goto exit;
+		}
+		status = psa_driver_wrapper_key_derivation_input_bytes(
+			&kdf_op, PSA_KEY_DERIVATION_INPUT_SECRET, V, hash_len);
+		if (status) {
+			goto exit;
+		}
+		status = psa_driver_wrapper_key_derivation_output_bytes(&kdf_op, operation->shared,
+									operation->shared_len);
+		if (status) {
+			goto exit;
+		}
+		psa_key_derivation_abort(&kdf_op);
```

Comment:
Someone might not know that PSA is inconsistent here and that abort is required on success.

Let's add a comment saying that abort is usually not required on success, but for KDF it is.

Reply:
the abort is to be able to run setup again on the next line. also there's no finish() for kdf, so you'd always have to call abort to end the operation.